relay/kusama/src/lib.rs

```diff
@@ -1927,6 +1927,211 @@ pub mod migrations {
+						}
+						expected_reserved_amounts.insert(
+							who,
+							current_reserved.saturating_sub(old_deposit - new_deposit),
+						);
+					} else {
+						// Nothing should change
+						expected_proxies.insert(who.clone(), (proxies, old_deposit));
+						expected_reserved_amounts.insert(who, current_reserved);
+					}
+				}
+
+				let pre_upgrade_state = (expected_proxies, expected_reserved_amounts);
+				Ok(pre_upgrade_state.encode())
+			}
+
+			fn on_runtime_upgrade() -> Weight {
+				let mut reads = 0u64;
+				let mut writes = 0u64;
+				let mut proxies_removed_total = 0u64;
+
+				Proxies::<Runtime>::translate(
+					|who: AccountId, (mut proxies, old_deposit): PrevProxiesValue<Runtime>| {
+						// Remove filter out IdentityJudgement proxies.
+						let proxies_len_before = proxies.len() as u64;
+						proxies
+							.retain(|proxy| proxy.proxy_type != PrevProxyType::IdentityJudgement);
+						let proxies_len_after = proxies.len() as u64;
+
+						let deposit = if proxies_len_before - proxies_len_after > 0 {
+							log::info!(
+								"Removing {} IdentityJudgement proxies for {:?}",
+								proxies_len_before - proxies_len_after,
+								&who
+							);
+							proxies_removed_total
+								.saturating_accrue(proxies_len_before - proxies_len_after);
+
+							let new_deposit =
+								pallet_proxy::Pallet::<Runtime>::deposit(proxies.len() as u32);
+
+							// Be kind and don't increase the deposit in case it increased (can
+							// happen if param change).
+							let deposit =
+								if new_deposit < old_deposit { new_deposit } else { old_deposit };
+							if deposit < old_deposit {
+								writes.saturating_inc();
+								<Balances as ReservableCurrency<AccountId>>::unreserve(
+									&who,
+									old_deposit - deposit,
+								);
+							}
+
+							deposit
+						} else {
+							// Nothing to do, use the old deposit.
+							old_deposit
+						};
+
+						reads.saturating_accrue(proxies_len_before + 1);
+						writes.saturating_accrue(proxies_len_after + 1);
+
+						// No need to keep the k/v around if there're no proxies left.
+						match proxies.is_empty() {
+							true => None,
+							false => Some((proxies, deposit)),
+						}
+					},
+				);
+
+				log::info!("Removed {} IdentityJudgement proxies in total", proxies_removed_total);
+				<Runtime as frame_system::Config>::DbWeight::get().reads_writes(reads, writes)
+			}
+
+			#[cfg(feature = "try-runtime")]
+			fn post_upgrade(state: Vec<u8>) -> Result<(), sp_runtime::TryRuntimeError> {
+				use frame_support::ensure;
+				use sp_runtime::TryRuntimeError;
+
+				let (expected_proxies, expected_total_reserved): (
+					BTreeMap<AccountId, PrevProxiesValue<Runtime>>,
+					BTreeMap<AccountId, Balance>,
+				) = Decode::decode(&mut &state[..]).expect("Failed to decode pre-upgrade state");
+
+				// Check Proxies storage is as expected
+				for (who, (proxies, deposit)) in Proxies::<Runtime>::iter() {
```

Comment:
Could use `iter_keys` in case some storage entry became on-decodable, but maybe thats just my OCD 😅

Reply:
`try-runtime` complains on any undecodable storage, which was how we realised we needed this migration to begin with :)